app/models/specialist_document.rb

```diff
@@ -81,30 +97,39 @@ def headers_list(headers)
     end
   end
 
-  def link?(facet, permitted_value)
-    facet["type"] == "text" &&
-      permitted_value.is_a?(Array) &&
-      facet["filterable"] == true
+  def allowed_value_facet_type(facet)
+    if %w[text nested].include?(facet["type"]) && facet["filterable"]
+      "link"
+    elsif %w[text nested].include?(facet["type"])
+      "preset_text"
+    else
+      facet["type"]
+    end
   end
 
-  def preset_text?(facet, permitted_value)
-    facet["type"] == "text" &&
-      permitted_value.is_a?(Array) &&
-      facet["filterable"] == false
+  def sub_facet_type(facet)
+    if %w[text nested].include?(facet["type"]) && facet["filterable"]
+      "sub_facet_link"
+    elsif %w[text nested].include?(facet["type"])
+      "sub_facet_text"
+    else
+      facet["type"]
+    end
   end
 
-  def allowed_value(allowed_values, metadata_facet_value)
-    allowed_values.select do |allowed_value|
-      next unless allowed_value["value"] == metadata_facet_value ||
-        metadata_facet_value.is_a?(Array) &&
-          allowed_value["value"].in?(metadata_facet_value)
+  def selected_allowed_values(allowed_values, metadata_facet_values)
+    [metadata_facet_values].flatten.map { |metadata_facet_value|
+      selected_allowed_value = allowed_values.detect { |allowed_value|
```

Comment:
Coverage report shows no coverage for this line when detect returns nil.

Reply:
added tests